File: software/examples/vbnet/ExampleThreshold.vb
Imports System
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Voltage/Current Bricklet 2.0

    ' Callback subroutine for power callback
    Sub PowerCB(ByVal sender As BrickletVoltageCurrentV2, ByVal power As Integer)
        Console.WriteLine("power: " + (power/1000.0).ToString() + " W")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim vc As New BrickletVoltageCurrentV2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register power callback to subroutine PowerCB
        AddHandler vc.PowerCallback, AddressOf PowerCB

        ' Configure threshold for power "greater than 10 W"
        ' with a debounce period of 1s (1000ms)
        vc.SetPowerCallbackConfiguration(1000, False, ">"C, 10*1000, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/voltage_current_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Voltage/Current Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	vc, _ := voltage_current_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	vc.RegisterPowerCallback(func(power int32) {
		fmt.Printf("power: %f W\n", float64(power)/1000.0)
	})

	// Configure threshold for power "greater than 10 W"
	// with a debounce period of 1s (1000ms).
	vc.SetPowerCallbackConfiguration(1000, false, '>', 10*1000, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
